android/src/com/unciv/app/MultiplayerTurnCheckWorker.kt

```diff
@@ -0,0 +1,210 @@
+package com.unciv.app
+
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import android.os.Build
+import androidx.core.app.NotificationCompat
+import androidx.core.app.NotificationManagerCompat
+import androidx.work.*
+import com.badlogic.gdx.backends.android.AndroidApplication
+import com.unciv.UncivGame
+import com.unciv.logic.GameInfo
+import com.unciv.models.translations.tr
+import com.unciv.ui.worldscreen.mainmenu.OnlineMultiplayer
+import java.util.*
+import java.util.concurrent.TimeUnit
+
+
+class MultiplayerTurnCheckWorker(appContext: Context, private val workerParams: WorkerParameters)
+    : Worker(appContext, workerParams) {
+
+    companion object {
+        const val WORK_DATA_FAIL_COUNT_INT = "WORK_DATA_FAIL_COUNT_INT"
+        const val WORK_DATA_GAME_ID = "WORK_DATA_GAME_ID"
+        const val WORK_TAG = "UNCIV_MULTIPLAYER_TURN_CHECKER_WORKER"
+        const val NOTIFICATION_CHANNEL_ID_INFO = "UNCIV_NOTIFICATION_CHANNEL_INFO"
+        const val NOTIFICATION_CHANNEL_ID_SERVICE = "UNCIV_NOTIFICATION_CHANNEL_SERVICE"
+        const val NOTIFICATION_ID_SERVICE = 1
+        const val NOTIFICATION_ID_INFO = 2
+
+        fun enqueue(appContext: Context, failCount: Int, delayInMinutes: Long, gameId: String) {
+            val inputData = workDataOf(WORK_DATA_GAME_ID to gameId, WORK_DATA_FAIL_COUNT_INT to failCount)
+
+            val constraints = Constraints.Builder()
+                    // If no internet is available, worker waits before becoming active.
+                    .setRequiredNetworkType(NetworkType.CONNECTED)
+                    .build()
+
+            val checkTurnWork = OneTimeWorkRequestBuilder<MultiplayerTurnCheckWorker>()
+                    .setConstraints(constraints)
+                    .setInitialDelay(delayInMinutes, TimeUnit.MINUTES)
+                    .setInputData(inputData)
+                    .addTag(WORK_TAG)
+                    .build()
+
+            WorkManager.getInstance(appContext).enqueue(checkTurnWork)
+        }
+
+        fun createNotificationChannelInfo(appContext: Context) {
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
```

Comment:
What is this check supposed to prevent?

Reply:
Notification channels were introduced in Android API level 26, which is `VERSION_CODES.O`

Reply:
Cool, please add this as a comment :)

Reply:
Added as a comment.